File: Backend/app/src/main/java/com/example/demo/repository/ChannelRepository.java
package com.example.demo.repository;

import com.example.demo.model.Channel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ChannelRepository extends JpaRepository<Channel, Long> {
    List<Channel> findByUserId(Long userId);
}

File: Mobile/app/src/main/java/com/example/myapplication/data/repository/ChannelRepository.kt
package com.example.myapplication.data.repository

import com.example.myapplication.data.api.RetrofitInstance
import com.example.myapplication.data.model.Channel

class ChannelRepository {
    suspend fun getChannelsByUser(): List<Channel> {
        return RetrofitInstance.channelApi.getChannelsByUser();
    }
}